app/views/course/assessment/answer/programming/_test_cases.html.slim

```diff
@@ -1,127 +1,50 @@
-- test_cases = question.test_cases
-- public_test_cases = test_cases.select(&:public_test?)
-- private_test_cases = test_cases.select(&:private_test?)
-- evaluation_test_cases = test_cases.select(&:evaluation_test?)
-- submission = answer.submission
 - auto_grading = answer.auto_grading ? answer.auto_grading.actable : nil
 - answer_test_results_hash = auto_grading ? \
     auto_grading.test_results.map { |result| [result.test_case, result] }.to_h : \
     {}
+- is_grader = can?(:grade, answer.answer)
+- test_cases_by_type = question.test_cases_by_type
```

Comment:
As spoken, might be better if we do a `group_by` directly on the `answer_test_results_hash` instead.